### DISCUSSION TITLE
Root Component Not Re-rendering

### DISCUSSION THREAD

**Discussion by andelkocvjetkovic** [TanStack/query#8235] [General] [answered]
Hello!

I'm encountering an issue where the Root component in my [CodeSandbox example](https://codesandbox.io/p/sandbox/youthful-fermi-wzsz7p) does not re-render when logout is called, even though the cache is cleared using` queryClient.clear();.`

Expected Behavior: I expected Root to re-render when logout clears the cache, causing the user state to reset. However, this doesn’t seem to happen, and Root stays in its current state.

Could you please help me understand why Root is not re-rendering in this scenario, or if there’s something I may need to adjust?

Thank you!

**Comment by TkDodo** [comment, marked as answer]
`queryClient.clear()`, same as `queryClient.removeQueries` does not inform currently rendered components. If you want that, you want `queryClient.resetQueries`. However, I’m not sure what that would achieve: You would logout, and then the root would re-render, in which case the login query runs again, which, idk, fails?

This seems more like an architectural problem. Usually, you’d have different routes for “public” areas (login, signup) and “protected” areas (after login). When the user logs out, I usually just call `queryClient.clear()` and `redirect('/login')`, because the `login` page doesn't use any queries...

**Reply by andelkocvjetkovic** [reply]
thank you for clarification
